index.js

```diff
@@ -88,19 +90,59 @@ export default class Carousel extends Component {
     if (this.state.childrenLength) {
       this._setUpTimer();
     }
+    // Set up pages but not content. Content will be set up via onLayout event.
+    this._setUpPages();
   }
 
   componentWillUnmount() {
     this._clearTimer();
   }
 
   componentWillReceiveProps(nextProps) {
-    let childrenLength = 0;
-    if (nextProps.children) {
-      childrenLength = nextProps.children.length ? nextProps.children.length : 1;
+    if (this.props.children !== nextProps.children) {
+        let childrenLength = 0;
+        if (nextProps.children) {
+          const length = React.Children.toArray(nextProps.children).length;
```

Comment:
`props.children` is considered opaque and no assumptions should be made about its structure. It should always be handled with the [React.Children class](https://facebook.github.io/react/docs/react-api.html#react.children).